css/tokens/index.js

```diff
@@ -34,39 +28,165 @@ async function createTokens() {
+	} catch (err) {
+		throw err;
+	}
+}
+
+/**
+ *
+ * @param {string[]} paths token file paths
+ */
+// Print a warning if a token file lacks the required sass-export-section comments.
+function checkFileComments(paths) {
+	paths.forEach(path => {
+		try {
+			fs.readFile(path, 'utf8', function read(err, result) {
+				if (err) {
+					throw err;
+				}
+				if (!result.includes('@sass-export-section')) {
+					console.log(`Warning: ${path} is missing @sass-export-section annotations.`);
+				}
+			});
+		} catch (err) {
+			throw new Error(`Problem reading token files: ${err}`);
+		}
+	});
+}
+
+/**
+ *
+ * @param {import('./types').SassExportCollection} collection collection of grouped tokens to be sorted
+ * @returns {import('./types').SassExportCollection}
+ */
+function getSortedOrder(collection) {
+	return Object.fromEntries(Object.entries(collection).sort());
+}
+
+/**
+ *
+ * @param {import('./types').SassExportTokens} tokens raw token data by group
+ * @returns {import('./types').SassExportCollection}
+ */
+function collectTokens(tokens) {
+	/** @type {import('./types').SassExportCollection} */
+	const collection = {};
+	/** @type {import('./types').SassExportCollection} */
+	for (const [parent, tokenValues] of Object.entries(tokens)) {
+		//Currently using sass-export-section annotations in the token files for grouping.
+		//Tokens without annotations will be combined in the variables array.
+		if (parent === 'variables') continue;
+
+		const collectedValues = tokenValues.reduce(
 			(
-				/** @type {{ [x: string]: any; }} */ all,
-				/** @type {{ name: string | number; }} */ source
+				/** @type {import('./types').SassExportCollection} */ all,
+				/** @type {import('./types').SassExportTokenItem} */ current
 			) => {
-				all[source.name] = {
-					...source,
-					tokens: {}
+				const tokenName = current.name;
+
+				/** @type {string | import('./types').SassExportTokenNestedItem} */
+				const values = current.mapValue
+					? { ...getNestedTokens(current) }[tokenName]
+					: convertBoolean(current.compiledValue);
+				all[parent] = {
+					...all[parent],
+					[tokenName]: values
 				};
 				return all;
 			},
 			{}
 		);
+		collection[parent] = {
+			name: parent,
+			location: `/css/src/tokens/${parent}.scss`,
+			tokens: collectedValues[parent]
+		};
+
+		collection.allTokens = { ...collection.allTokens, ...collectedValues[parent] };
+	}
+
 	return collection;
 }
 
+/**
+ *
+ * @param {import('./types').SassExportTokenItem} child
+ * @param {import('./types').SassExportTokenItem} [parent]
+ * @returns {{[key: string]: string | boolean | import('./types').SassExportTokenNestedItem}}
+ */
+function getNestedTokens(child, parent) {
+	if (!child.mapValue) {
+		const { name, compiledValue, value } = child;
+		let newCompiledValue;
+
+		//For nested maps, the compiled value for each key is retrieved by parsing the parent's compiledValue string.
+		if (!compiledValue) {
+			newCompiledValue = parent?.compiledValue
```

Comment:
Parsing the parent here, in the child, means that we're parsing the exact same thing many times over.

Each time a child property on `light` theme is parsed, the whole parent is parsed again. Should we clean this up to ensure we're not doing duplicate work?

Reply:
In fact, it looks to me like traversing all the child nodes may not be entirely necessary. The following riff on your code smacks them all into an object very similar (the same?) to the one being output:

```js
const parsedParent = parent?.compiledValue
.replace(/^\(/, '')
.replace(/\)$/, '')
.split(/\,(?![^(]*\))/)
.reduce((collection, val) => {
	const [name, value] = val.replaceAll(' ', '').replaceAll('"', '').split(':');
	collection[name] = convertBoolean(value);
	return collection;
}, {});

Reply:
values of `parsedParent` is 

```js
{
  text: "#171717",
  "text-subtle": "#505050",
  "text-invert": "#fff",
  "text-glow-high-contrast": "#171717",
  "box-shadow-light": "rgba(0,0,0,0.11)",
  "box-shadow-medium": "rgba(0,0,0,0.13)",
  "box-shadow-heavy": "rgba(0,0,0,0.18)",
  "box-shadow-extra-heavy": "rgba(0,0,0,0.22)",
  overlay: "rgba(0,0,0,0.7)",
  "overlay-invert": "#fff",
  "body-background": "#fff",
  "body-background-medium": "#f2f2f2",
  "alternate-background": "#171717",
  "alternate-background-medium": "#2f2f2f",
  border: "#e6e6e6",
  "table-header": "#e6e6e6",
  "table-row-header": "#fafafa",
  "table-border-dark": "#d2d2d2",
  "hover-base": "#2f2f2f",
  "hover-invert": "#fafafa",
  "border-white-high-contrast": "rgba(255,255,255,0)",
  "border-yellow-high-contrast": "rgba(0,0,0,0)",
  "code-header": "#e6e6e6",
  "code-block": "#f2f2f2",
  "inline-code": "#e6e6e6",
  "code-highlight-background": "#fff4ce",
  visited: "#624991",
  "score-low-off": "rgba(168,0,0,0.3)",
  "score-low": "#b62626",
  "score-medium-off": "rgba(255,187,0,0.3)",
  "score-medium": "#ffb900",
  "score-high-off": "rgba(72,157,72,0.3)",
  "score-high": "#2a8b2a",
  hyperlink: "#0065b3",
  "primary-base": "#0078d4",
  "primary-background": "#d7eaf8",
  "primary-background-glow-high-contrast": "#d7eaf8",
  "primary-dark": "#004173",
  "primary-hover": "#0065b3",
  "primary-active": "#00579a",
  "primary-box-shadow": "rgba(0,101,179,0.3)",
  "primary-invert": "#fff",
  "secondary-base": "#a2a2a2",
  "secondary-background": "#e6e6e6",
  "secondary-background-glow-high-contrast": "#e6e6e6",
  "secondary-dark": "#2f2f2f",
  "secondary-hover": "#d2d2d2",
  "secondary-active": "#d2d2d2",
  "secondary-box-shadow": "rgba(0,0,0,0.3)",
  "secondary-invert": "#000",
  "tertiary-base": "#243a5e",
  "tertiary-background": "#061329",
  "tertiary-background-glow-high-contrast": "#061329",
  "tertiary-dark": "#a4b2c9",
  "tertiary-hover": "#14294c",
  "tertiary-active": "#70819f",
  "tertiary-box-shadow": "rgba(38,23,63,0.3)",
  "tertiary-invert": "#fff",
  "success-base": "#107c10",
  "success-background": "#dff6dd",
  "success-background-glow-high-contrast": "#dff6dd",
  "success-dark": "#054b16",
  "success-hover": "#0b6413",
  "success-active": "#054b16",
  "success-box-shadow": "rgba(72,157,72,0.3)",
  "success-invert": "#fff",
  "info-base": "#8661c5",
  "info-background": "#efd9fd",
  "info-background-glow-high-contrast": "#efd9fd",
  "info-dark": "#3b2e58",
  "info-hover": "#8661c5",
  "info-active": "#624991",
  "info-box-shadow": "rgba(32,24,67,0.3)",
  "info-invert": "#fff",
  "warning-base": "#ffb900",
  "warning-background": "#fff4ce",
  "warning-background-glow-high-contrast": "#fff4ce",
  "warning-dark": "#6a4b16",
  "warning-hover": "#d19501",
  "warning-active": "#966802",
  "warning-box-shadow": "rgba(255,187,0,0.3)",
  "warning-invert": "#000",
  "danger-base": "#a80000",
  "danger-background": "#fde7e9",
  "danger-background-glow-high-contrast": "#fde7e9",
  "danger-dark": "#470001",
  "danger-hover": "#870000",
  "danger-active": "#630001",
  "danger-box-shadow": "rgba(168,0,0,0.3)",
  "danger-invert": "#fff",
  "facepile-red": "#a80000",
  "facepile-teal": "#328e9f",
  "facepile-blue": "#0078d4",
}

```

Reply:
Pushed the latest version that doesn't require passing parent.